tooling/test/sugar.go

```diff
@@ -162,6 +165,34 @@ func (e ExpectBuilder) Status(statusCode int) ExpectBuilder {
 	return e
 }
 
+func (e ExpectBuilder) StatusMatch(pattern string) ExpectBuilder {
+	re := regexp.MustCompile(`^(\d+)(x+)$`)
+	matches := re.FindStringSubmatch(pattern)
+	if len(matches) != 3 {
+		panic("invalid status pattern")
+	}
+
+	// Extract the leading digits and the number of 'x' characters
+	leadingDigits := matches[1]
+	numXs := len(matches[2])
+
+	// Compute the lower bound
+	from, err := strconv.Atoi(leadingDigits + strings.Repeat("0", numXs))
+	if err != nil {
+		panic(fmt.Sprintf("invalid status pattern: %v", err))
+	}
+
+	// Compute the upper bound
+	to, err := strconv.Atoi(leadingDigits + strings.Repeat("9", numXs))
+	if err != nil {
+		panic(fmt.Sprintf("invalid status pattern: %v", err))
+	}
+
+	e.StatusCodeFrom_ = from
+	e.StatusCodeTo_ = to
+	return e
+}
+
 func (e ExpectBuilder) StatusBetween(from, to int) ExpectBuilder {
```

Comment:
Are we using this function elsewhere? @lidel Are there any cases where we would want to do `501-503` or other?

Reply:
I don't think so, in practice we care about specific code, or 4XX or 5XX